src/expr/src/scalar/func/impls/mz_timestamp.rs

```diff
@@ -128,6 +129,18 @@ sqlfunc!(
     }
 );
 
+sqlfunc!(
+    #[sqlname = "date_to_mz_timestamp"]
+    #[is_monotone = true]
+    fn cast_date_to_mz_timestamp(a: Date) -> Result<Timestamp, EvalError> {
+        let ts = CheckedTimestamp::try_from(NaiveDate::from(a).and_hms_opt(0, 0, 0).unwrap())?;
```

Comment:
Would it be better to replace this with `super::CastDateToTimestamp(None).call(a)`?
E.g. calling the `CastDateToTimestamp` directly from here, rather than re-implementing some of that logic. Didn't see any precedent of doing that, so wasn't sure best practice.

Reply:
I'd say both approaches are fine, but I have a slight preference for your current version. (For example, a minor thing that might possibly go wrong with the `super::CastDateToTimestamp(None).call(a)` version is that maybe we add some error msg there that specifically mentions `CastDateToTimestamp`, and then we'd confusingly show that error msg also here if somebody doesn't realize that it's also called here.)

Reply:
That's a really good point, thanks